Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Revolver.cs

```diff
@@ -130,7 +130,7 @@ public bool TryRevolverInsert(EntityUid revolverUid, RevolverAmmoProviderCompone
                 return false;
             }
 
-            for (var i = Math.Min(ev.Ammo.Count - 1, component.Capacity - 1); i >= 0; i--)
+            for (var i = 0; i < component.Capacity; i++)
             {
                 var index = (component.CurrentIndex + i) % component.Capacity;
```

Comment:
Doesn't that entirely duplicate lines 171-193 at this point.

Reply:
Furthermore what 133-158 is doing is looking through every slot from the initial position in an attempt to load multiple rounds from a container (the Speedloader), and breaking when it runs out of ammo. It grabs round types of the speedloader (which can be mixed types) and takes it out of the speedloader.

What 171-193 is doing is looking through every slot from the initial position to insert a single round from the hand and then instantly finishing.